Hotfix/create endturn feature

KENUBBE opened the pull request:
Hotfix for createEndturnFeature() 
* Added opponent player variable to track currentPlayer and opponentPlayer

KENUBBE committed:
Deleted target

KENUBBE committed:
Added test for opponent player in endTurn feature

KENUBBE committed:
Hotfix for createEndturnFeature() -> Added opponent player variable to track currentPlayer and opponentPlayer

KENUBBE committed:
Created randomizer in GameEngine to determine player to start and togglePlayer

Juulis commented on src/main/java/models/GameEngine.java:
```diff
@@ -11,15 +12,17 @@ public GameEngine(){
     private Deck deck;
     private boolean game;
     private boolean playing;
+    private Random rand;
```
Finns det någon anledning att rand är en klassvariabel?

KENUBBE replied:
Because I can!!!

Nä men absolut, den tar upp onödigt minne i heapen😝I fix for you!

Juulis commented on src/main/java/models/GameEngine.java:
```diff
@@ -77,26 +80,33 @@ public void initPlayer() {
         }
     }
 
+    public Player getOpponentPlayer(){
```
🤒 😷     🔥 🔥 🔥 

😆 🤣

mohammedalmahfoodh replied:
I think you missed range values[1,2] int randomNr = rand.nextInt();

Juulis replied:
mohammed har rätt, range saknas! :P

KENUBBE committed:
Synced with dev

KENUBBE committed:
Rewrote old tests for endTurn() and getStartingPlayer() to use spy instead -> Added randomizer as dependency instead of instance variable

KENUBBE commented:
* Rewrote old tests for endTurn() and getStartingPlayer() to use spy instead. Because of that I was able to remove "int randomNr" as an argument from getStartingPlayer().

* Added randomizer as a dependency instead of a instance variable, so that we can use it wherever we need to.

Juulis commented on src/main/java/models/GameEngine.java:
```diff
@@ -77,26 +78,38 @@ public void initPlayer() {
-    public void getPlayerToStart(int randomNr) {
+    public void getPlayerToStart() {
+        Random rand = makeRandom();
+        int randomNr = rand.nextInt();
```
what range?

KENUBBE committed:
Fixed random range in getStartingPlayer()

KENUBBE committed:
Changes in test code for endTurn(), showTable() and determinePlayerToStart()
, to pass!

KENUBBE committed:
Update GameEngineTest.java

Removed repeatedtest 100 to just @Test
